source/src/protocols/motifs/LigandDiscoverySearch.cc

```diff
@@ -292,694 +366,1790 @@ utility::vector1<core::conformation::ResidueOP> LigandDiscoverySearch::get_all_r
 
-			//utility::vector1< core::Size > atom_vector;
-			if ( ligresOP->atom_is_hydrogen(atom_i) ) { continue; }
-			// This is a for loop to iterate over each atom's connected atoms:
-			core::conformation::Residue::AtomIndices atom_i_connects(  ligresOP->bonded_neighbor( atom_i ) );
-			for ( core::Size atom_j = 1; atom_j <= atom_i_connects.size(); ++ atom_j ) {
-				if ( ligresOP->atom_is_hydrogen(atom_i_connects[atom_j]) ) { continue; }
-				// This is the next for loop to find connects for the second atom, giving us the final atom number (atom k)
-				core::conformation::Residue::AtomIndices atom_j_connects(  ligresOP->bonded_neighbor( atom_i_connects[atom_j] ) );
-				for ( core::Size atom_k = 1; atom_k <= atom_j_connects.size(); ++ atom_k ) {
-					if ( ligresOP->atom_is_hydrogen(atom_j_connects[atom_k]) ) { continue; }
-					chemical::AtomType atom_i_type(ligresOP->atom_type(atom_i));
-					if ( atom_i != atom_j_connects[atom_k] ) {
+// @brief this function takes in a selected scorefunctionOP and gets the ddg of the selected poseOP (ideally with a placed ligand), and returns the ddg
+core::Real LigandDiscoverySearch::get_pose_ddg(core::scoring::ScoreFunctionOP score_fxn, core::pose::PoseOP & my_pose)
+{
+	//use the passed score function to get interface deltas on the working_pose
+	std::map< std::string, core::Real > interface_mapX_postdock = protocols::ligand_docking::get_interface_deltas('2', *my_pose, score_fxn, "");
 
-						//make the 3 atom vector
-						utility::vector1< utility::vector1< core::Size > > cur_motif_indices;
+	//return the ddg, which is interface_delta_2 in the interface map
+	return interface_mapX_postdock["interface_delta_2"];
+}
 
-						utility::vector1< core::Size > atom_i_vector;
-						atom_i_vector.push_back( atom_i );
-						atom_i_vector.push_back( atset->atom_type_index( ligresOP->atom_type(atom_i).atom_type_name() ) );
+// @brief This function adds a ligand mutableresiduetypeOP to the residue type set for working_pose and original_pose. This occurs on the first instance of this ligand passing enough filters to be used in highresdock
+// the ligand is added to both so that the ligand can be a part of the original_pose for future iterations of placement attempts for the ligand, as well as the working_pose for immediate use
+void LigandDiscoverySearch::add_ligand_to_pose_residuetypeset(const core::chemical::MutableResidueTypeOP lig_mrt)
+{
+	//code to add type set of imported ligand into pose
+	core::chemical::PoseResidueTypeSetOP rts( working_pose_->conformation().modifiable_residue_type_set_for_conf( core::chemical::FULL_ATOM_t ) );
 
-						utility::vector1< core::Size > atom_j_vector;
-						atom_j_vector.push_back( atom_i_connects[atom_j] );
-						atom_j_vector.push_back( atset->atom_type_index( ligresOP->atom_type(atom_i_connects[atom_j]).atom_type_name() ) );
+	//make a ResidueTypeSetCOP that will be pulled from the PoseResidueTypeSetOP
+	core::chemical::ResidueTypeSetCOP def_rts(rts->default_rts());
```

Comment:
I wouldn't bother to grab the default RTS -- you can check the PoseRTS for the presence of the ligand directly. (Which is likely more what you want, as if the PoseRTS has the ligand but the underlying typeset doesn't, then you probably still want to skip adding it.)